src/gsy_e/models/market/two_sided.py

```diff
@@ -300,11 +300,20 @@ def accept_bid_offer_pair(self, bid: Bid, offer: Offer, clearing_rate: float,
 
     def _get_offer_from_seller_origin_id(self, seller_origin_id):
         """Get the first offer that has the same seller_origin_id."""
+        if seller_origin_id is None:
+            # Many offers may have seller_origin_id=None; Avoid looking for them as it is
```

Comment:
I found this by accident! As the `seller_origin_id` is None by default, if someone forgets to fill this parameter, then it will remain None. Now, if that happens for many times in the same market, which is very probable, then bids/offers will get matched randomly and no validation will stop that trade.
I'm not sure whether it really happens or not, but I added that `if` statement just in case.

Reply:
So you actually did never experience this, right? 
I think we need to make the `seller_origin_id`, 'seller_origin'  and `seller_id` required parameters for the Offer object and here we could add an assertion:
```python
assert seller_origin_id is not None, "seller_origin_id should never be None"
```
(the same applies to the buyer* members in the Bid object).
Tagging @spyrostz for his opinion.

Reply:
I did experience this when I was writing unit tests. But I didn't see such a thing in other use cases.

Reply:
I suggest that we open a new ticket for that as it may affect other parts of the code. What do you think?

Reply:
Hi @hannesdiedrich ,
https://gridsingularity.atlassian.net/browse/GSYE-312